ui/lib/core/addon/components/confirm-action.hbs

```diff
@@ -3,57 +3,49 @@
-    @verticalPosition={{this.verticalPosition}}
-    @onOpen={{this.toggleConfirm}}
-    @onClose={{this.toggleConfirm}}
-    as |d|
+{{#if @isInDropdown}}
+  <Hds::Dropdown::ListItem::Interactive
+    @text={{@buttonText}}
+    @color={{@buttonColor}}
+    {{on "click" (fn (mut this.showConfirmModal) true)}}
+    ...attributes
+  />
+{{else}}
+  <Hds::Button
+    data-test-confirm-action-trigger
+    @text={{@buttonText}}
+    @color={{@buttonColor}}
+    {{on "click" (fn (mut this.showConfirmModal) true)}}
+    ...attributes
+  />
+{{/if}}
+
+{{#if this.showConfirmModal}}
+  <Hds::Modal
+    id="confirm-action-modal"
+    @color={{or @modalColor "critical"}}
+    @size="small"
+    @onClose={{fn (mut this.showConfirmModal) false}}
+    as |M|
   >
-    <d.Trigger
-      @htmlTag="button"
-      class={{concat @buttonClasses " popup-menu-trigger" (if d.isOpen " is-active")}}
-      type="button"
-      disabled={{this.disabled}}
-      data-test-confirm-action-trigger="true"
-    >
-      {{yield}}
-      {{#if (eq @buttonClasses "toolbar-link")}}
-        <Chevron @direction={{if this.showConfirm "up" "down"}} data-test-confirm-action-chevron />
-      {{/if}}
-    </d.Trigger>
-    <d.Content @defaultClass="popup-menu-content">
-      <div class="box confirm-action-message">
-        <div class="message is-highlight">
-          <div class="message-title" data-test-confirm-action-title>
-            <Icon @name="alert-triangle-fill" />
-            {{this.confirmTitle}}
-          </div>
-          <p>
-            {{this.confirmMessage}}
-          </p>
-        </div>
-        <div class="confirm-action-options">
-          {{! TODO Hds::Button replacement - skipping because modal will replace this confirm inline-popup menu }}
-          <button
-            type="button"
-            disabled={{or this.disabled this.isRunning}}
-            class="link is-destroy"
-            data-test-confirm-button="true"
-            {{on "click" (fn this.onConfirm d.actions)}}
-          >
-            {{#if this.isRunning}}
-              <span class="loader is-inline-block"></span>
-            {{else}}
-              {{this.confirmButtonText}}
-            {{/if}}
-          </button>
-          <button type="button" class="link" data-test-confirm-cancel-button="true" {{on "click" d.actions.close}}>
-            {{this.cancelButtonText}}
-          </button>
-        </div>
-      </div>
-    </d.Content>
-  </BasicDropdown>
-</div>
\ No newline at end of file
+    <M.Header data-test-confirm-action-title @icon="alert-diamond">
+      {{or @confirmTitle "Are you sure?"}}
+    </M.Header>
+    <M.Body>
+      {{this.confirmMessage}}
+    </M.Body>
+    <M.Footer as |F|>
+      <Hds::ButtonSet>
+        <Hds::Button
+          data-test-confirm-button
+          disabled={{or @disabled @isRunning}}
+          @icon={{if @isRunning "loading"}}
+          @color={{if (or (not @modalColor) (eq @modalColor "critical")) "critical" "primary"}}
```

Comment:
I feel like this could be simplified if we have `this.modalColor` getter which defaults to `critical` (and we could use on line 26 too)

Reply:
Yeah! I went back and forth on this quite a bit with how to manage the args for this component. Will make this a getter!